library/ssl_client.c

```diff
@@ -988,8 +989,37 @@ int mbedtls_ssl_write_client_hello( mbedtls_ssl_context *ssl )
                                                                 buf_len,
                                                                 msg_len ) );
         mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_SERVER_HELLO );
```

Comment:
If offering early data and if the middlebox compatibility mode is enabled, we need to send a CCS immediately after the first ClientHello, probably through a MBEDTLS_SSL_CLIENT_CCS_AFTER_CLIENT_HELLO state like in the prototype.